native/src/main/java/tech/sourced/ParserDemo.java

```diff
@@ -2,33 +2,46 @@
 
 import com.ansorgit.plugins.bash.lang.BashVersion;
 import com.ansorgit.plugins.bash.lang.lexer.BashLexer;
+import com.ansorgit.plugins.bash.lang.parser.BashParser;
 import com.ansorgit.plugins.bash.lang.parser.BashParserDefinition;
 import com.ansorgit.plugins.bash.lang.parser.BashPsiBuilder;
 import com.ansorgit.plugins.bash.lang.parser.FileParsing;
 import com.intellij.core.CoreApplicationEnvironment;
 import com.intellij.core.CoreProjectEnvironment;
 import com.intellij.lang.ASTNode;
+import com.intellij.lang.ParserDefinition;
 import com.intellij.lang.PsiBuilder;
 import com.intellij.lang.impl.PsiBuilderFactoryImpl;
 import com.intellij.mock.MockProject;
 import com.intellij.openapi.Disposable;
+import com.intellij.psi.tree.IElementType;
+
+import java.util.Properties;
 
 public class ParserDemo {
+    private static final String debugKey = "bashsupport.debug"; // from BashParser
+
     public static void run(CharSequence code, BashVersion version) {
-        BashPsiBuilder b = builder(code, version);
-        FileParsing fp = new FileParsing();
+        boolean debug = "true".equals(System.getProperty(debugKey)) || "true".equals(System.getenv(debugKey));
+        System.out.println("debug = " + debug);
+
+        MockProject project = project();
```

Comment:
This is an obsession of mine, but I recommend declaring every variable, method argument or class field as `final`, unless it requires to be reassigned for a good reason.